### PR TITLE
Add disposal of NpgsqlDataSource after opening connection

### PR DESCRIPTION
**Pull request by marcominerva**
## Motivation and Context (Why the change? What's the scenario?)
See https://github.com/microsoft/kernel-memory/issues/679.

## High level description (Approach, Design)
If I call `ImportDocumentAsync` and then `AskAsync`, after the execution I see that two sessions are still in the list:

image

This PR adds the disposal the **NpgsqlDataSource** after each connection. In this way, no pending sessions remain when the execution completes.

~~NOTE: for the moment, I have left the `IDisposable` interface on **PostgresDbClient.cs** (with an empty implementation for reference), but actually it is no longer necessary, as explained in https://github.com/microsoft/kernel-memory/issues/679#issuecomment-2202303865. So, I think it could be safely removed.~~

### CODE REVIEW

File: extensions/Postgres/Postgres/Internals/PostgresDbClient.cs [hunk 1]
```diff
@@ -685,7 +684,8 @@ private async Task<NpgsqlConnection> ConnectAsync(CancellationToken cancellation
     {
         try
         {
-            return await this._dataSource.OpenConnectionAsync(cancellationToken).ConfigureAwait(false);
+            using var dataSource = this._dataSourceBuilder.Build();
```
Comment (dluc) [inline extensions/Postgres/Postgres/Internals/PostgresDbClient.cs:687]: should use the async dispose right? ie `await using`

File: extensions/Postgres/Postgres/Internals/PostgresDbClient.cs [hunk 2]
```diff
@@ -28,7 +28,7 @@ internal sealed class PostgresDbClient : IDisposable
     private const string PgErrDatabaseDoesNotExist = "3D000"; // invalid_catalog_name
 
     private readonly ILogger _log;
-    private readonly NpgsqlDataSource _dataSource;
+    private readonly NpgsqlDataSourceBuilder _dataSourceBuilder;
```
Comment (dluc) [inline extensions/Postgres/Postgres/Internals/PostgresDbClient.cs:31]: by removing `NpgsqlDataSource` we can also remove `IDisposable` (line 22) and the `Dispose` methods

File: extensions/Postgres/Postgres/Internals/PostgresDbClient.cs [hunk 3]
```diff
@@ -228,8 +228,8 @@ public async Task CreateTableAsync(
             if (await this.DoesTableExistAsync(origInputTableName, cancellationToken).ConfigureAwait(false))
             {
                 // Check if the custom SQL contains the lock placeholder (assuming it's not commented out)
-                bool missingLockStatement = (!string.IsNullOrEmpty(this._createTableSql)
-                                             && !this._createTableSql.Contains(PostgresConfig.SqlPlaceholdersLockId, StringComparison.Ordinal));
+                bool missingLockStatement = !string.IsNullOrEmpty(this._createTableSql)
+                                             && !this._createTableSql.Contains(PostgresConfig.SqlPlaceholdersLockId, StringComparison.Ordinal);
```
Comment (dluc) [inline extensions/Postgres/Postgres/Internals/PostgresDbClient.cs:232]: nit: indentation

**Review by dluc — CHANGES_REQUESTED** [review]

File: extensions/Postgres/Postgres/Internals/PostgresDbClient.cs (hunk 2 above)
Comment (marcominerva) [reply extensions/Postgres/Postgres/Internals/PostgresDbClient.cs:31]: I have removed the `IDsposable` interface from `NpgsqlDataSource `, that in turn makes the `IDisposable` interface on `PostgresMemory` on longer necessary.

**Review by dluc — APPROVED** [review]

**Pull request merged**